### PR TITLE
Add log and tanh to Num::Grad::Variable

### LINKED ISSUE #82: `log` and `tanh` not defined for `Num::Grad::Variable`
I'm working on some toy models using `num.cr` while following some PyTorch examples. As I was setting up the forward calculations using a gradient variable, I needed `log` and `tanh` functions which are currently not implemented.

### PR DESCRIPTION
**Pull request by nogginly**
Fixes #82 

@christopherzimmerman, this PR adds `log` and `tanh` to `Num::Grad::Variable` with backprop using their derivatives. 
* Both CPU and OCL are supported.
* Test cases are included

### CODE REVIEW

File: src/grad/backends/cpu.cr [hunk 1]
```diff
@@ -59,6 +59,18 @@ module Num::Grad
     [r0]
   end
 
+  # :nodoc:
+  def log_backward(
+    gradient : Tensor(U, CPU(U)),
+    a : Variable(Tensor(U, CPU(U)))
+  ) : Array(Tensor(U, CPU(U))) forall U
+    r0 = gradient.map(a.value) do |i, j|
+      # puts "#{i}, #{j} -> #{i} * #{1 / j} "
```
Comment (christopherzimmerman) [inline src/grad/backends/cpu.cr:68]: This should be cleaned up

File: src/tensor/backends/opencl/kernels/custom.cr [hunk 2]
```diff
@@ -286,6 +286,54 @@ end
 
 create_kernel_children(ExpBackwardsKernel, [Float32, Float64])
 
+# :nodoc:
+class Num::LogBackwardsKernel(T) < Num::Kernel(T)
```
Comment (christopherzimmerman) [inline src/tensor/backends/opencl/kernels/custom.cr:290]: Looks awesome, thanks for doing the CL work as well!

**Review by christopherzimmerman — CHANGES_REQUESTED** [review]
@nogginly just one debug statement to clean up and should be good to merge, thanks for the PR!

File: src/grad/backends/cpu.cr (hunk 1 above)
Comment (nogginly) [reply src/grad/backends/cpu.cr]: Done!
